### PR TITLE
Update docs

### LINKED ISSUE #150: docs with BemEntityName

### PR DESCRIPTION
**Pull request by blond**
Resolved #150

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,130 +65,64 @@ According to original BEM-naming convention it looks like the following:
...
+* Element's simple modifier — `block-name__elem_mod`.
 
-BEM entities can be defined with a help of JS object with the following fields:
+### Delimiters
 
-* `block` — a block name. The field is required because only a block exists as an independent BEM entity
-* `elem` — an element name.
-* `modName` — a modifier name.
-* `modVal` — a modifier value.
+In the original naming uses the following delimiters:
```
Comment (tadatuta) [inline README.md]: In the original -> The original

File: README.md [hunk 2]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
 
-* `block` — a block name. The field is required because only a block exists as an independent BEM entity
-* `elem` — an element name.
-* `modName` — a modifier name.
-* `modVal` — a modifier value.
+The original naming uses the following delimiters:
 
-The modifier consists of a pair of fields `modName` and `modVal`. This means that the field `modVal` without `modName` has no meaning.
+* `__` — to separate element from block
+* `_` — to separate modifier name from block or element and to separate value of modifier from name of modifier
```
Comment (krishnevsky) [inline README.md]: to separate a modifier name from a block or element and to separate a modifier value from a modifier name

File: README.md [hunk 3]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
-    modVal: 'val'
-}
-```
-
-To describe the boolean modifier field `modVal` must be specified as `true`.
+Common misconceptions
+---------------------
 
-Example:
+BEM methodology involves the use of flat structure inside a block. It means that BEM entity can not be represented as an element of the other element and the following string representation will be invalid:
```
Comment (krishnevsky) [inline README.md]: The BEM methodology uses a flat structure inside blocks. This means that a BEM entity can't be represented as an element of another element, ...

File: README.md [hunk 4]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
+For more information, see the [FAQ](https://en.bem.info/methodology/faq/):
 
-* [parse(str)](#parsestr)
-* [stringify(obj)](#stringifyobj)
-* [elemDelim](#elemdelim)
-* [modDelim](#moddelim)
-* [modValDelim](#modvaldelim)
-
-### parse(str)
+> [Why does BEM not recommend using elements within elements (block__elem1__elem2)?](https://en.bem.info/methodology/faq/#why-does-bem-not-recommend-using-elements-within-elements-block__elem1__elem2)
```
Comment (krishnevsky) [inline README.md]: Why doesn't BEM recommend...

File: README.md [hunk 5]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
 
-It parses string into naming object.
-
-Example:
+Also there is no such BEM entity as a modifier and an element modifier simultaneously so the following string representation will be invalid:
```
Comment (krishnevsky) [inline README.md]: Also, a BEM entity can't be a modifier and an element modifier simultaneously,

File: README.md [hunk 6]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
 ```js
-const bemNaming = require('bem-naming');
-
-bemNaming.parse('block__elem_mod_val');  // { block: 'block', elem: 'elem',
-                                         //   modName: 'mod', modVal: 'val' }
+'block_block-mod-name_block-mod-val__elem-name_elem-mod-name_elem-mod-val'
 ```
 
-### stringify(obj)
+Convention by Harry Roberts
```
Comment (krishnevsky) [inline README.md]: Harry Roberts' naming convention

File: README.md [hunk 7]
```diff
@@ -59,136 +69,70 @@ According to original BEM-naming convention it looks like the following:
...
-```js
-'block_block-mod-name_block-mod-val__elem-name_elem-mod-name_elem-mod-val'
+// ➜ block__elem--mod
 ```
 
 Custom naming convention
 ------------------------
 
-Use `bemNaming` function to create instance to manage naming of your own naming convention.
+Use [bemNaming](#bemnaming-elem-mod-wordpattern-) function to create instance to manage naming of your own naming convention.
```
Comment (krishnevsky) [inline README.md]: Use the [bemNaming](#bemnaming-elem-mod-wordpattern-) function to create an instance where you can manage your own naming convention.

File: README.md [hunk 8]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+* [modDelim](#moddelim)
+* [modValDelim](#modvaldelim)
 
-This option can take object with following fields:
+### bemNaming({ elem, mod, wordPattern })
 
-* `name` — separates name of modifier from blocks and elements.
+Parameter     | Type     | Description                                                                | Default
+--------------|----------|----------------------------------------------------------------------------|---------
+`elem`        | `string` | Separates element's name from block.                                       | `__`
```
Comment (krishnevsky) [inline README.md]: Separates an element's name from a block.

File: README.md [hunk 9]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+`mod`         | `string`, `{ name: string, val: string }` | Separates modifier from block or element. | `_`
```
Comment (krishnevsky) [inline README.md]: Separates a modifier from a block or an element.

File: README.md [hunk 10]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+`mod.name`    | `string` | Separates name of modifier from block or element.                          | `_`
```
Comment (krishnevsky) [inline README.md]: Separates a modifier name from a block or an element.

File: README.md [hunk 11]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+`mod.val`     | `string` | Separates value of modifier from name of modifier.                         | Default as the value of the `mod.name`.
```
Comment (krishnevsky) [inline README.md]: Separates the value of a modifier from the modifier name.

File: README.md [hunk 12]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+`wordPattern` | `string` | Defines which symbols can be used for block, element and modifier's names. | `[a-z0-9]+(?:-[a-z0-9]+)*`
```
Comment (krishnevsky) [inline README.md]: Defines which characters can be used in names of blocks, elements, and modifiers.

File: README.md [hunk 13]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
-  Default as `_`.
+### parse(str)
 
-* `val` — separates value of modifier from name of modifier.
+Parameter | Type     | Description
+----------|----------|--------------------------------
+`str`     | `string` | BEM entity name representation.
 
-  Default as the value of the `name`.
+It parses string into `BemEntityName` instance.
```
Comment (krishnevsky) [inline README.md]: Parses the string into an instance of `BemEntityName`.

File: README.md [hunk 14]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
----------------------------
+### stringify(entityName)
 
-According to this convention elements are delimited with two underscores (`__`), modifiers are delimited by two hyphens (`--`), and values of modifiers are delimited by one underscore (`_`).
+Parameter    | Type                      | Description
+-------------|---------------------------|--------------------------------
+`entityName` | `BemEntityName`, `object` | BEM entity name representation.
 
-Read more in the [Guidelines](http://cssguidelin.es/#bem-like-naming).
+It forms a string according to `BemEntityName` instance.
```
Comment (krishnevsky) [inline README.md]: Forms a string from the instance of `BemEntityName`.

File: README.md [hunk 15]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
-    modName: 'mod'
+    mod: { name: 'mod', val: 'val' }
 });
+
+// ➜ block__elem_mod_val
 ```
 
+### elemDelim
+
+String to separate element from block.
```
Comment (krishnevsky) [inline README.md]: String to separate an element from a block.

File: README.md [hunk 16]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+
+### modDelim
+
+String to separate modifier name from block or element.
```
Comment (krishnevsky) [inline README.md]: String to separate a modifier name from a block or element.

File: README.md [hunk 17]
```diff
@@ -203,76 +147,92 @@ const myNaming = bemNaming({
...
+
+### modValDelim
+
+String to separate value of modifier from name of modifier.
```
Comment (krishnevsky) [inline README.md]: String to separate a modifier value from the name of the modifier.

File: README.md [hunk 18]
```diff
@@ -30,26 +31,35 @@ Usage
...
 Table of Contents
 -----------------
 
+* [Object representation](#object-representation)
 * [String representation](#string-representation)
-* [Object representation of BEM entity](#object-representation-of-bem-entity)
-* [API](#api)
 * [Common misconceptions](#common-misconceptions)
-* [Custom naming convention](#custom-naming-convention)
 * [Convention by Harry Roberts](#convention-by-harry-roberts)
```
Comment (godfreyd) [inline README.md]: @tadatuta  мы же вроде от Гарри отказались?

File: README.md (hunk 6 above)
Comment (godfreyd) [inline README.md]: @tadatuta туда же. Мы выпиливали из документации Гарри.

File: README.md (hunk 18 above)
Comment (blond) [inline README.md]: @godfreyd ради него этот пакет изначально и делался :)

И нет, не отказались. Хотя мы его нигде и не использовали, чтобы отказываться.

**Comment by blond** [comment]
🆙

**Pull request merged**